src/Lecture1.hs

```diff
@@ -1,126 +1,147 @@
-
->>> lastDigit 42
-2
->>> lastDigit (-17)
-7
-
-🕯 HINT: use the @mod@ function
-
--}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
-
-{- | Write a function that takes three numbers and returns the
-difference between the biggest number and the smallest one.
-
->>> minmax 7 1 4
-6
-
-Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
-and 1 is the smallest, and 7 - 1 = 6.
-
-Try to use local variables (either let-in or where) to implement this
-function.
--}
-minmax x y z = error "TODO"
-
-{- | Implement a function that takes a string, start and end positions
-and returns a substring of a given string from the start position to
-the end (including).
-
->>> subString 3 7 "Hello, world!"
-"lo, w"
-
->>> subString 10 5 "Some very long String"
-""
-
-This function can accept negative start and end position. Negative
-start position can be considered as zero (e.g. substring from the
-first character) and negative end position should result in an empty
-string.
--}
-subString start end str = error "TODO"
-
-{- | Write a function that takes a String — space separated numbers,
-and finds a sum of the numbers inside this string.
-
->>> strSum "100    -42  15"
-73
-
-The string contains only spaces and/or numbers.
--}
-strSum str = error "TODO"
-
-{- | Write a function that takes a number and a list of numbers and
-returns a string, saying how many elements of the list are strictly
-greater than the given number and strictly lower.
-
->>> lowerAndGreater 3 [1 .. 9]
-"3 is greater than 2 elements and lower than 6 elements"
-
-Explanation: the list [1 .. 9] contains 9 elements: [1, 2, 3, 4, 5, 6, 7, 8, 9]
-The given number 3 is greater than 2 elements (1 and 2)
-and lower than 6 elements (4, 5, 6, 7, 8 and 9).
-
-🕯 HINT: Use recursion to implement this function.
--}
-lowerAndGreater n list = error "TODO"
+-- | Implement a function that takes two numbers and finds sum of
+-- their squares.
+--
+-- >>> sumOfSquares 3 4
+-- 25
+--
+-- >>> sumOfSquares (-2) 7
+-- 53
+--
+-- Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
+-- is 25.
+sumOfSquares :: Integer -> Integer -> Integer
+sumOfSquares x y = x ^ 2 + y ^ 2
+
+-- | Implement a function that returns the last digit of a given number.
+--
+-- >>> lastDigit 42
+-- 2
+-- >>> lastDigit (-17)
+-- 7
+--
+-- 🕯 HINT: use the @mod@ function
+lastDigit :: Integer -> Integer
+lastDigit n = if n < 0 then mod (0 - n) 10 else mod n 10
+
+-- | Write a function that takes three numbers and returns the
+-- difference between the biggest number and the smallest one.
+--
+-- >>> minmax 7 1 4
+-- 6
+--
+-- Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
+-- and 1 is the smallest, and 7 - 1 = 6.
+--
+-- Try to use local variables (either let-in or where) to implement this
+-- function.
+minmax x y z = (tripletMax x y z) - (tripletMin x y z)
+  where
+    tripletMax x y z = max (max x y) (max y z)
```

Comment:
You can slightly simplify this function to call `max` one less time:

```haskell
tripletMax x y z = max (max x y) z
```

The same goes for `tripleMin`